cmd/debos/debos.go

```diff
@@ -226,17 +243,9 @@ func main() {
 		}
 	}
 
-	for _, a := range r.Actions {
-		err = a.Run(&context)
-
-		// This does not stop the call of stacked Cleanup methods for other Actions
-		// Stack Cleanup methods
-		defer a.Cleanup(&context)
-
-		// Check the state of Run method
-		if exitcode = checkError(&context, err, a, "Run"); exitcode != 0 {
-			return
-		}
+	exitcode = do_run(r, &context)
```

Comment:
Just a nitpick: may I ask you to compact this a bit? Something like.
if exitcode = do_run(r, &context); exitcode !=0 {
  return 0;
}